source/fundamentals/builders/sort.txt

```diff
@@ -11,3 +11,321 @@ Sorts Builders
+
+   collection.find().sort(Sorts.ascending("<field name>"))
+
+This will return a 
+:java-docs:`FindIterable </apidocs/mongodb-driver-sync/com/mongodb/client/FindIterable.html>`
+object containing the documents in your collection sorted 
+from smallest to largest on the specified field name. 
+
+.. example::
+
+   In this code example, we use ``ascending`` to
+   sort the ``sort_example`` collection by ``"_id"``:
+
+   .. code-block:: java
+   
+      import static com.mongodb.client.model.Sorts.ascending;
+   
+      // <MongoCollection setup code here>
+      
+      Bson id_sort = ascending("_id");
+      List<Document> results = new ArrayList<>();
+      collection.find().sort(id_sort).into(results);
+      for (Document result : results) {
+          System.out.println(result.toJson());
+      }
+
+   The output of the above code snippet should look something like this: 
+   
+   .. code-block:: none
+   
+      {"_id": 1, "letter": "c", "food": "coffee with milk"}
+      {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+      {"_id": 3, "letter": "a", "food": "maple syrup"}
+      ...
+   
+An important thing for you to keep in mind is that MongoDB does
+not guarantee sort order for documents that have fields with identical values.
+This means that ``ascending("letter")`` could have any of these
+documents in the first position: 
+
+.. code-block:: none
+
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+   {"_id": 2, "letter": "a", "donuts and coffee"}
+
+If you would like to have a guaranteed sort order 
+for documents that have fields with identical values
+you can specify additional fields to sort
+on in the event of a tie. 
+
+We can specify an ``ascending`` sort on ``"letter"`` followed
+by ``"_id"`` as follows:
+
+.. code-block:: java
+
+   import static com.mongodb.client.model.Sorts.ascending;
+
+   // <MongoCollection setup code here>
+   
+   collection.find().sort(ascending("letter", "_id"))
+
+The above code snippet returns the documents in the
+``sort_example`` collection in the following order: 
+
+.. code-block:: none
+
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+
+
+Descending
+~~~~~~~~~~
+
+To specify a descending sort, use the ``Sorts.descending()``
+static factory method. A descending sort returns your documents
+from largest to smallest on the field name specified.
+
+The following code snippet shows how to specify a ``descending``
+sort on the ``"_id"`` field: 
+
+.. code-block:: java
+   
+   import static com.mongodb.client.model.Sorts.descending;
+
+   // <MongoCollection setup code here>
+   
+   collection.find().sort(descending("_id"))
```

Comment:
this is missing a trailing semicolon, applies to all instances

Reply:
This would not compile ^

Reply:
Thanks for the note. Fixed missing semicolons in code snippets here: https://github.com/mongodb/docs-java/pull/59/commits/78d43cbe0a1a8c61713462c626f97a7160350f0d

Going forward, I will try to better check code snippets to make sure they will compile in given language. Will possibly look into bluehawk to better separate code snippets from docs content (let IDE catch bugs and then auto replace text on page).